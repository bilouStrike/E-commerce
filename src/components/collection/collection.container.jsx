import WithSpinner from './../withSpinner/withSpinner.component';
import Collection from './collection.component';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { selectIsCollectionsLoading } from './../../redux/shop/shop-selectors';

const mapStateToProps = state => ({
    isLoading: !selectIsCollectionsLoading(state)
});

const collectionsContainer = compose(
    connect(mapStateToProps),
    WithSpinner
)(Collection)

export default collectionsContainer;