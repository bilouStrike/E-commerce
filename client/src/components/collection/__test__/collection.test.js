import { shallow } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import { CollectionPage } from '../collection.component';

it('Should render Collection component', () =>{
    const mockCollection = {
        title: 'hats',
        items: [{
            id:2,
            imageUrl: 'url',
            price: '250',
            name: 'name1',
            quantity: 2
            }]
    }
    expect(toJson(shallow(<CollectionPage collection={mockCollection}/>))).toMatchSnapshot();
});
