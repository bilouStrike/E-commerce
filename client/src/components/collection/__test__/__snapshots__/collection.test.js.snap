// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Collection component 1`] = `
<div
  className="collection-page"
>
  <h2
    className="title"
  >
    hats
  </h2>
  <div
    className="items"
  >
    <CollectionItem
      item={
        Object {
          "id": 2,
          "imageUrl": "url",
          "name": "name1",
          "price": "250",
          "quantity": 2,
        }
      }
      key="2"
    />
  </div>
</div>
`;
